cli: `Test{SQLLex,RunQuery}` fail with `TESTFLAGS=-v`

Looks like this code gets confused https://github.com/cockroachdb/cockroach/blob/4e6ec545cc65dc0051a797996e9642b4b5e8bdc2/pkg/cli/sql.go#L1046-L1048

```
$ make test PKG=./pkg/cli TESTFLAGS=-v TESTS=SQLLex
...
=== RUN   TestSQLLex
test logs left over in: /tmp/logTestSQLLex831856190
--- FAIL: TestSQLLex (0.13s)
	test_log_scope.go:81: test logs captured to: /tmp/logTestSQLLex831856190
	test_log_scope.go:62: use -show-logs to present logs inline
	sql_test.go:151: 
		select '
		\?
		;
		';
		:
		expected: +---------------+
		| e'\n\\?\n;\n' |
		+---------------+
		| ␤             |
		| \?␤           |
		| ;␤            |
		|               |
		+---------------+
		(1 row)
		
		got: +---------------+
		| e'\n\\?\n;\n' |
		+---------------+
		| ␤             |
		| \?␤           |
		| ;␤            |
		|               |
		+---------------+
		(1 row)
		
		Time: 7.969132ms
		
FAIL
exit status 1
FAIL	github.com/cockroachdb/cockroach/pkg/cli	0.167s
make: *** [test] Error 1
```

--- a/pkg/cli/cli_test.go
+++ b/pkg/cli/cli_test.go
@@ -140,6 +140,18 @@ func newCLITest(params cliTestParams) cliTest {
 	return c
 }
 
+// setCLIDefaultsForTests invokes initCLIDefaults but pretends the
+// output is not a terminal, even if it happens to be. This ensures
+// e.g. that tests ran with -v have the same output as those without.
+func setCLIDefaultsForTests() {
+	initCLIDefaults()
+	cliCtx.terminalOutput = false
+	cliCtx.showTimes = false
+	// Even though we pretend there is no terminal, most tests want
+	// pretty tables.
+	cliCtx.tableDisplayFormat = tableDisplayPretty
+}
+
 // stopServer stops the test server.
 func (c *cliTest) stopServer() {
 	if c.TestServer != nil {

--- a/pkg/cli/context.go
+++ b/pkg/cli/context.go
@@ -60,9 +60,11 @@ func initCLIDefaults() {
 	// other client commands are non-interactive, regardless of whether
 	// the standard input is a terminal.
 	cliCtx.isInteractive = false
+	// See also setCLIDefaultForTests() in cli_test.go.
 	cliCtx.terminalOutput = isatty.IsTerminal(os.Stdout.Fd())
 	cliCtx.tableDisplayFormat = tableDisplayTSV
 	if cliCtx.terminalOutput {
+		// See also setCLIDefaultForTests() in cli_test.go.
 		cliCtx.tableDisplayFormat = tableDisplayPretty
 	}
 	cliCtx.showTimes = false

--- a/pkg/cli/sql_test.go
+++ b/pkg/cli/sql_test.go
@@ -98,8 +98,7 @@ select '''
 		},
 	}
 
-	// Some other tests (TestDumpRow) mess with this, so make sure it's set.
-	cliCtx.tableDisplayFormat = tableDisplayPretty
+	setCLIDefaultsForTests()
 
 	// We need a temporary file with a name guaranteed to be available.
 	// So open a dummy file.

--- a/pkg/cli/sql_util_test.go
+++ b/pkg/cli/sql_util_test.go
@@ -98,10 +98,9 @@ func TestRunQuery(t *testing.T) {
 	conn := makeSQLConn(url.String())
 	defer conn.Close()
 
-	// Use a buffer as the io.Writer.
-	var b bytes.Buffer
+	setCLIDefaultsForTests()
 
-	cliCtx.tableDisplayFormat = tableDisplayPretty
+	var b bytes.Buffer
 
 	// Non-query statement.
 	if err := runQueryAndFormatResults(conn, &b, makeQuery(`SET DATABASE=system`)); err != nil {

